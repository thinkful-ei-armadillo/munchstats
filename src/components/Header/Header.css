header{
  padding: 10px;
  text-align: center;
  background: #0E4749;
  color:#EFE7DA;
  font-size: 20px;
  box-shadow:1px 0 6px grey;
  height:90px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

header a:visited {
  color:#EFE7DA;
}

header h1{
  margin: 0;
}


header a:hover{
  color: #000000;
}

@media only screen and (max-width: 400px) {
  header {
    flex-direction: column;
    height:120px;
  }
}