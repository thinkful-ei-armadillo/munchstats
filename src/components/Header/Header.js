import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import UserContext from '../../contexts/UserContext';
import './Header.css';

export default class Header extends Component {
  static contextType = UserContext;

  handleLogoutClick = () => {
    this.context.processLogout();
  };

  renderLogoutLink() {
    return (
      <div>
        <Link
          to='/profile'
          style={{textDecoration: 'none'}}>
          {this.context.user.name}
        </Link>
        <nav>
          <Link
            onClick={this.handleLogoutClick}
            to='/login'
            style={{textDecoration: 'none'}}>
            Logout
          </Link>
        </nav>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <nav>
        <Link
          to='/login'
          style={{textDecoration: 'none'}}>
          Log In
        </Link>
        {' | '}
        <Link
          to='/register'
          style={{textDecoration: 'none'}}>
          Sign Up
        </Link>
      </nav>
    );
  }

  render() {
    return (
      <header className="backgroundColor2 textColor2">
        <h1>
          <Link
            to='/'
            style={{textDecoration: 'none'}}>
            Munch Stats
          </Link>
        </h1>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </header>
    );
  }
}

