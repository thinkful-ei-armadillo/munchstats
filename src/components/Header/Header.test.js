import React from 'react';
import Header from './Header';
import {MemoryRouter} from 'react-router-dom';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Header />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><Header /></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Header/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});