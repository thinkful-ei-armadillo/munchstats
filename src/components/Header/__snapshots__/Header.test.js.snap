// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the UI as expected 1`] = `
<header
  className="backgroundColor2 textColor2 shadow"
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <img
      alt="Munch Stats"
      className="logoTest"
      src="logo_m.png"
    />
  </a>
  <nav>
    <a
      href="/login"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Log In
    </a>
     | 
    <a
      href="/register"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Sign Up
    </a>
  </nav>
</header>
`;
